src/openforms/emails/service.py

```diff
@@ -0,0 +1,56 @@
+from django_yubin.models import Message
+
+from openforms.submissions.models import Submission
+
+from .constants import (
+    X_OF_CONTENT_TYPE_HEADER,
+    X_OF_CONTENT_UUID_HEADER,
+    X_OF_EVENT_HEADER,
+    EmailContentTypeChoices,
+    EmailEventChoices,
+)
+
+__all__ = ["get_last_confirmation_email"]
+
+
+def get_last_confirmation_email(submission: Submission) -> tuple[str, int] | None:
+    """
+    Get the last created confirmation email from the
+    :class:`django_yubin.models.Message` model.
+
+    :return: A tuple of the HTML content and message ID of the last created confirmation
+      email for the specified submission. ``None`` if there weren't any.
+    """
+    # We only care about messages that were created after the submission
+    messages = Message.objects.filter(date_created__gte=submission.created_on).order_by(
+        "-date_created"
+    )
+
+    for message in messages:
+        email = message.get_email_message()
+
+        has_submission = (
+            email.extra_headers.get(X_OF_CONTENT_TYPE_HEADER)
+            == EmailContentTypeChoices.submission
+        )
+        if not has_submission:
+            continue
```

Comment:
you can trigger this by using `@override_settings` to pin the email backend to django-yubin in tests (django automatically patches this to the in memory backend) and use a plain `django.core.email.send_mail` for some bogus email that should end up in the table as well. Then just make sure to include our "magic markers" in the email body to get false positives and verify they're filtered out.

Orrr... this check may even not be relevant at all because of the checks we perform later anyway to correlate the message and submission :thinking: